libgalois/include/katana/Properties.h

```diff
@@ -298,9 +298,7 @@ class BooleanPropertyReadOnlyView {
   using value_type = uint8_t;
 
   static Result<BooleanPropertyReadOnlyView> Make(
-      const arrow::BooleanArray& array) {
-    return BooleanPropertyReadOnlyView(array);
-  }
+      const arrow::BooleanArray& array);
```

Comment:
Single line functions are better defined in the header so that they _can_ be inlined by the compiler. Does this not compile for host code using nvcc?

Reply:
No, it doesn't compile with `nvcc`. My earlier measurements show that non-inline functions often outperform inline functions likely because the branch prediction kicks in for non-inline functions. Doesn't `g++`/`clang` have whole program optimization / link-time code generation option that would still make inline the functions defined not in headers?